main/webapp/modules/core/scripts/views/data-table/data-table-view.js

```diff
@@ -178,10 +177,10 @@ DataTableView.prototype._renderPagingControls = function(pageSizeControls, pagin
     previousPage.addClass("inaction");
   }
 
-  lastPageSpan = $('<span>').attr("id", "viewpanel-paging-last").text(self._lastPageNumber);
-  pageControlSpan = $('<span>').attr("id", "viewpanel-paging-current");
+  var pageControlsSpan = $('<span>').attr("id", "viewpanel-paging-current");
   
-  currentPageInput = $('<input type="number">')
+  var pageInputSize = 20 + (8 * ui.dataTableView._lastPageNumber.toString().length);
+  var currentPageInput = $('<input type="number">')
     .change(function(evt) { self._onChangeGotoPage(this, evt); })
```

Comment:
I think **something** like this for example would be more efficient... 
```suggestion
    /* Don't fire the GotoPage until the user has stopped typing or pressing keys...
    
    // .change(function(evt) { self._onChangeGotoPage(this, evt); })
    // .keydown(function(evt) { self._onKeyDownGotoPage(this, evt); })
    
   // So, we check the keyup, and delay for some configurable Timeout, default .7 secs, and then attempt to GotoPage...
    .bind('keyup', function(evt) {self._onChangeGotoPage(this, evt); })
```

REFS: 
1. https://stackoverflow.com/questions/1909441/how-to-delay-the-keyup-handler-until-the-user-stops-typing
2. https://github.com/OpenRefine/OpenRefine/blob/3ca08f6ff189cc1bd806a1d49d2ead538882c0e2/main/webapp/modules/core/scripts/facets/text-search-facet.js#L221

Reply:
@thadguidry: Are you sure this is needed/useful?

I'm not sure I understand you. You want to get the change of page to auto-execute after a certain period of time? What is the added benefit? Not pressing [RETURN]?

Regards,
   Antoine

Reply:
Lets drop that need however.  Automatically hitting [RETURN] for folks.  They get that with a mouse click on the existing Prev / Next buttons.  I was thinking of accessibility as other apps work this way by holding down up or down arrow to go up or down to pages.  Bit it currently doesn't feel fluid to me with this current code change for some reason.  So let's park that need for now.

The input box should require the user to press [RETURN] key to go to the page they have typed in.  DONE.
If the users want to cycle forward or backward, they will have to use mouse clicks on our Prev / Next buttons.  DONE.

If you want to revert those last changes for now, that's fine.  And then we can merge.

Reply:
> Lets drop that need however. Automatically hitting [RETURN] for folks. 

If the user presses [RETURN] after changing the number, the browser does the out-of-focus (blur) automatically. I didn't code anything for that.

>They get that with a mouse click on the existing Prev / Next buttons. I was thinking of accessibility as other apps work this way by holding down up or down arrow to go up or down to pages. But it currently doesn't feel fluid to me with this current code change for some reason. So let's park that need for now.

Are you talking of the Arrow Keys, or the Arrow button? ![image](https://user-images.githubusercontent.com/12217525/82954127-15df3e80-9f7a-11ea-84ed-a125227bb291.png))

Maybe if you give me a precise example of an application and the place in the UI that handles that, I could understand better.

> The input box should require the user to press [RETURN] key to go to the page they have typed in. DONE.
> If the users want to cycle forward or backward, they will have to use mouse clicks on our Prev / Next buttons. DONE.

And don't forget the Keyboard Arrows! That's a nice add-on you proposed.

> If you want to revert those last changes for now, that's fine. And then we can merge.

K. I'll wrap up and finish this issue.

Regards,
   Antoine